js/src/events/updates/SendUpdateModal.js

```diff
@@ -0,0 +1,72 @@
+import React from 'react'
+
+import {ModalForm} from '../../forms/Form'
+
+
+const EVENT_EMAIL_UPDATE_FIELDS = [
+  {name: 'subject', required: true},
+  {name: 'message', required: true, type: 'md', max_length: 255000}
+]
+const MESSAGE_NAME_TT_SLUG = 'message_to_ticket_type__'
+
+
+export class SendUpdateModal extends React.Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      initialValue: {status: props.event.status},
+      fields: this.buildFieldList()
+    }
+  }
+
+  componentDidUpdate (prevProps) {
+    if (prevProps.ticketTypes !== this.props.ticketTypes)
+      this.setState({fields: this.buildFieldList()})
+  }
+
+  buildFieldList () {
+    if (this.props.ticketTypes && this.props.ticketTypes.length > 1) {
+      const messages = this.props.ticketTypes
+        .map(tt => Object.assign({}, EVENT_EMAIL_UPDATE_FIELDS[1], {
+          name: MESSAGE_NAME_TT_SLUG + tt.id,
+          title: `Message to ${tt.name} ticket holders`,
+          required: false
+        }))
+      return [EVENT_EMAIL_UPDATE_FIELDS[0]].concat(messages)
+    } else {
+      return JSON.parse(JSON.stringify(EVENT_EMAIL_UPDATE_FIELDS))
+    }
+  }
+
+  dataFormatter (formData) {
+    return Object.keys(formData).reduce((data, k) => {
```

Comment:
I'm not exactly sure why this is using reduce, but I suspect it can be written in a cleaner way.